src/main/java/ua/com/conductor/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package ua.com.conductor.service.impl;
+
+import java.util.Optional;
+import ua.com.conductor.exception.AuthenticationException;
+import ua.com.conductor.lib.Inject;
+import ua.com.conductor.lib.Service;
+import ua.com.conductor.model.User;
+import ua.com.conductor.service.AuthenticationService;
+import ua.com.conductor.service.UserService;
+import ua.com.conductor.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> optionalUser = userService.findByEmail(email);
+        if (optionalUser.isEmpty()) {
+            throw new AuthenticationException("Incorrect login");
```

Comment:
let's not make the life of some hackers easier) 
Let's not refactor this method the way so we are throwing AuthenticationException only once and only with the message: 
"Incorrect login or password"; for example you can user optionalUser.isPresent() with &&